File: WebView/WebChromeClient/CustomView/MainActivity.java
package com.example.myapplication;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.view.View;
import android.webkit.WebChromeClient;
import android.webkit.WebView;
import android.widget.FrameLayout;

public class MainActivity extends AppCompatActivity {

    private WebView webView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        /**/
        initViews();
        /**/
        initWebChromeClient();
    }

    private void initViews() {
        webView = findViewById(R.id.webView);
    }

    private void initWebChromeClient() {
        webView.setWebChromeClient(new WebChromeClient() {

            private View mCustomView = null;
            private CustomViewCallback mCustomViewCallback = null;
            protected FrameLayout mFullscreenContainer = null;
            private int mOriginalOrientation = 0;
            private int mOriginalSystemUiVisibility = 0;

            @Nullable
            @Override
            public Bitmap getDefaultVideoPoster() {
                if (mCustomView == null) {
                    return null;
                } else {
                    return BitmapFactory.decodeResource(getApplicationContext().getResources(), 2130837573);
                }
            }

            @Override
            public void onHideCustomView() {
                ((FrameLayout)getWindow().getDecorView()).removeView(this.mCustomView);
                this.mCustomView = null;
                getWindow().getDecorView().setSystemUiVisibility(this.mOriginalSystemUiVisibility);
                setRequestedOrientation(this.mOriginalOrientation);
                this.mCustomViewCallback.onCustomViewHidden();
                this.mCustomViewCallback = null;
            }

            @Override
            public void onShowCustomView(View paramView, CustomViewCallback paramCustomViewCallback) {
                if (this.mCustomView != null) {
                    onHideCustomView();
                    return;
                }
                this.mCustomView = paramView;
                this.mOriginalSystemUiVisibility = getWindow().getDecorView().getSystemUiVisibility();
                this.mOriginalOrientation = getRequestedOrientation();
                this.mCustomViewCallback = paramCustomViewCallback;
                ((FrameLayout)getWindow().getDecorView()).addView(this.mCustomView, new FrameLayout.LayoutParams(-1, -1));
                getWindow().getDecorView().setSystemUiVisibility(3846 | View.SYSTEM_UI_FLAG_LAYOUT_STABLE);
            }
            
        });
    }
}

File: WebView/WebChromeClient/CustomView/MainActivity.kt
package com.example.myapplication

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.os.Bundle
import android.view.View
import android.webkit.WebChromeClient
import android.webkit.WebView
import android.widget.FrameLayout
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var webView: WebView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        /**/
        initViews()
        /**/
        initWebChromeClient()
    }

    private fun initViews() {
        webView = findViewById(R.id.webView)
    }

    private fun initWebChromeClient() {
        webView.webChromeClient = object : WebChromeClient() {

            private var mCustomView: View? = null
            private var mCustomViewCallback: CustomViewCallback? = null
            protected var mFullscreenContainer: FrameLayout? = null
            private var mOriginalOrientation: Int = 0
            private var mOriginalSystemUiVisibility: Int = 0

            override fun getDefaultVideoPoster(): Bitmap? {
                return if (mCustomView == null) {
                    null
                } else BitmapFactory.decodeResource(applicationContext.resources, 2130837573)
            }

            override fun onHideCustomView() {
                (window.decorView as FrameLayout).removeView(this.mCustomView)
                this.mCustomView = null
                window.decorView.systemUiVisibility = this.mOriginalSystemUiVisibility
                requestedOrientation = this.mOriginalOrientation
                this.mCustomViewCallback!!.onCustomViewHidden()
                this.mCustomViewCallback = null
            }

            override fun onShowCustomView(paramView: View, paramCustomViewCallback: CustomViewCallback) {
                if (this.mCustomView != null) {
                    onHideCustomView()
                    return
                }
                this.mCustomView = paramView
                this.mOriginalSystemUiVisibility = window.decorView.systemUiVisibility
                this.mOriginalOrientation = requestedOrientation
                this.mCustomViewCallback = paramCustomViewCallback
                (window.decorView as FrameLayout).addView(this.mCustomView, FrameLayout.LayoutParams(-1, -1))
                window.decorView.systemUiVisibility = 3846
            }

        }
    }
}

